src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/profiler/ProfilerEditingTarget.java

```diff
@@ -393,10 +486,125 @@ public void onError(ServerError error)
             {
                globalDisplay_.showErrorMessage("Failed to Open Profile",
                      error.getMessage());
+               
+               continuation.execute(null);
             }
          });
    }
+   
+   private void persistHtmlPath(String htmlPath)
+   {
+      HashMap<String, String> props = new HashMap<String, String>();
+      props.put("htmlPath", htmlPath);
+      
+      sourceServer_.modifyDocumentProperties(
+         doc_.getId(),
+         props,
+         new SimpleRequestCallback<Void>("Error")
+         {
+            @Override
+            public void onResponseReceived(Void response)
+            {
+            }
+
+            @Override
+            public void onError(ServerError error)
+            {
+               globalDisplay_.showErrorMessage("Failed to Save Profile Properties",
+                     error.getMessage());
+            }
+      });
+   }
+
+   private void saveNewFile(final String suggestedPath,
+                            final Command executeOnSuccess)
```

Comment:
No, grabbed this one from the boilerplate we have but didn't end up using it, changed!